src/main/java/de/muenchen/isi/domain/service/calculation/InfrastrukturbedarfService.java

```diff
@@ -0,0 +1,773 @@
+                    .multiply(
+                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr16NachErsterstellung()
+                    )
+                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
+            personen.add(
+                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
+            );
+            yearToCalculate++;
+        }
+        if (yearToCalculate < numberOfYearsToCalculate) {
+            anzahlPersonenGesamt =
+                wohneinheiten
+                    .getWohneinheiten()
+                    .multiply(
+                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr17NachErsterstellung()
+                    )
+                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
+            personen.add(
+                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
+            );
+            yearToCalculate++;
+        }
+        if (yearToCalculate < numberOfYearsToCalculate) {
+            anzahlPersonenGesamt =
+                wohneinheiten
+                    .getWohneinheiten()
+                    .multiply(
+                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr18NachErsterstellung()
+                    )
+                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
+            personen.add(
+                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
+            );
+            yearToCalculate++;
+        }
+        if (yearToCalculate < numberOfYearsToCalculate) {
+            anzahlPersonenGesamt =
+                wohneinheiten
+                    .getWohneinheiten()
+                    .multiply(
+                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr19NachErsterstellung()
+                    )
+                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
+            personen.add(
+                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
+            );
+            yearToCalculate++;
+        }
+        if (yearToCalculate < numberOfYearsToCalculate) {
+            anzahlPersonenGesamt =
+                wohneinheiten
+                    .getWohneinheiten()
+                    .multiply(
+                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr20NachErsterstellung()
+                    )
+                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
+            personen.add(
+                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
+            );
```

Comment:
Kann man das nicht kürzer zusammenfassen, da immer derselbe Code aufgelistet wird. Und dafür am Anfang nur eine Überprüfung gegen den Wert der Variable macht?
Unterscheidet sich ja nur an den Wert X jeweils
`sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr X NachErsterstellung()`

Reply:
done.

Die Berechnungslogik innerhalb der If-Statements ist in eine eigene Methode ausgelagert.